README.md

```diff
@@ -557,9 +557,12 @@ If you'd like `just` to default to listing the recipes in the `justfile`, you ca
 
 ```make
 default:
-  @just --list
+  @# list recipes from the current justfile
+  @just --justfile {{justfile()}} --list
 ```
 
+Note that the `--justfile {{justfile()}}` above is needed to make sure the listed recipes are from the current `justfile`. Without it, if you executed `just -f /some/distant/justfile` or `just -f ./non-standard-justfile`, the plain `just --list` call in the `default` recipe would not use the file you provided. It would try to find a justfile in your current path, maybe even resulting in a `No justfile found` error.
```

Comment:
@casey Is this better?

Reply:
Is `just -f /some/distant/justfile` a problem? I think just will switch its working directory and find the justfile.

Reply:
in the case of that `.j` alias, the directory is already set. I don't think there's other solution to combine both (the alias and the list recipe, I mean).

Reply:
A more generic feature that `just` may need is an alternative to this [`$(MAKE) $(MFLAGS)`](https://www.gnu.org/software/make/manual/html_node/Options_002fRecursion.html) for calling a "sub" `just`.

Maybe, a `just()` function could be equivalent to `{{just_executable()}} --justfile {{justfile()}}`. Then, recommend that any recursive just calls are made with `{{just()}}` instead of plainly `just`.

Reply:
We definitely need something like. I want some way of running a just command in the middle of a recipe, but we haven't figured out the syntax yet.

Reply:
I'm having trouble reproducing the behavior you mentioned. I created a justfile in foo/.justfile, which has `just --list` in the first recipe, calling `just -f foo/.justfile` seems to work, it successfully calls just and lists the recipes in the justfile.

Reply:
The output below may help.

Notice that:
- the `jj` alias includes `-d .` option;
- when `pwd` is outside of `~`, I get a `No justfile found` _coming from the default recipe_;
- when `pwd` is inside my home (e.g. `~/src/`), `jj` works as intended.
- when `pwd` is inside my home, but inside some directory with another justfile, it lists recipes from that directory instead.

```
$ /bin/cat ~/justfile
default:
  @just --list

pwd:
  pwd

$ type jj
jj is aliased to `just -f ~/justfile -d .'

$ jj pwd
pwd
/usr/local

$ jj
error: No justfile found
error: Recipe `default` failed on line 2 with exit code 1

$ cd ~/src
$ jj
Available recipes:
    default
    pwd

$ cd ~/src/jp/just/
$ jj
Available recipes:
    build
    build-book
    changes                   # add git log messages to changelog
    check
    c                         # alias for `check`
    …
```